ec/resources/waf/rate_rule.go

```diff
@@ -283,19 +251,57 @@ func ResourceRateRuleRead(
 	d.Set("name", resp.Name)
 	d.Set("num", resp.Num)
 	d.Set("keys", resp.Keys)
-	flattenedConditionGroups := FlattenConditionGroups(resp.ConditionGroups)
+	flattenedConditionGroups := flattenConditionGroups(resp.ConditionGroups)
 	d.Set("condition_group", flattenedConditionGroups)
 	return diags
 }
 
-func ResourceRateRuleUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func ResourceRateRuleUpdate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	config := m.(**api.ClientConfig)
+	wafService, err := buildWAFService(**config)
 
-	var diags diag.Diagnostics
-	return diags
-}
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ruleID := d.Id()
+	rule, err := expandRateRule(d)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	log.Printf(
+		"[INFO] Updating WAF Rate Rule %s for Account >> %s",
+		ruleID,
+		rule.CustomerID)
+	log.Printf("[DEBUG] Customer ID: %+v \n", rule.CustomerID)
+	log.Printf("[DEBUG] Disabled: %+v\n", rule.Disabled)
+	log.Printf("[DEBUG] DurationSec: %+v\n", rule.DurationSec)
+	log.Printf("[DEBUG] Name: %+v\n", rule.Name)
+	log.Printf("[DEBUG] Num: %+v\n", rule.Num)
+	log.Printf("[DEBUG] Keys: %+v\n", rule.Keys)
+	log.Printf("[DEBUG] ConditionGroups: %+v\n", rule.ConditionGroups)
+
+	resp, err := wafService.UpdateRateRule(*rule, ruleID)
+
+	if err != nil {
+		d.SetId("")
```

Comment:
Similarly here, would an API error result in setting the empty ID?

Reply:
Again, correct. Thanks for catching this.